Build failure on Ubuntu 26.04 LTS/pixi with GCC 15.2.0

## Problem description

Building Cantera from source fails during the SCons configuration stage with an error indicating that `std::isnan` is not functioning correctly. The issue occurs before compilation begins, during compiler feature detection.

The failure was observed using Ubuntu 26.04 with GCC 15.2.0 and a Pixi-managed Python 3.14 environment.

## Steps to reproduce

1. Clone Cantera from GitHub:

```bash id="d21pmx"
git clone git@github.com:Cantera/cantera.git
cd cantera
```

2. Create and activate the Pixi environment:

```bash id="4lv0kq"
pixi install
pixi shell
```

3. Build Cantera using SCons:

```bash id="rtrbyd"
scons build -j8
```

4. Observe the configuration failure:

```text id="njlwm0"
ERROR: Cantera requires a working implementation of 'std::isnan'.
If you have specified '-ffast-math' or equivalent as an optimization option,
either remove this option or add the '-fno-finite-math-only' option.
```

## Behavior

The build fails during SCons configuration checks even though no explicit floating-point optimization flags (e.g., `-ffast-math`) were set by the user.

Expected behavior:

* Cantera should either build successfully, or
* provide additional diagnostic information identifying the compiler flags or conditions that triggered the failure.

Observed behavior:

* The configuration stage detects an invalid `std::isnan` implementation and aborts before compilation begins.

## System information

* Cantera version: 4.0.0a0 main | git commit `4be08e69b`
* OS: Ubuntu 26.04 LTS
* Python version: 3.14 
* Pixi version: 0.68.1
* SCons version: 4.10.1

### Compiler information

Compiler details:

```text id="q5df6x"
Target: x86_64-linux-gnu
Thread model: posix
gcc version 15.2.0 (Ubuntu 15.2.0-16ubuntu1)
```

No compiler flag environment variables are set:

```text id="aqaz8v"
$ env | grep -E '^(CC|CXX|CFLAGS|CXXFLAGS|CPPFLAGS|LDFLAGS)='
[no output]
```

## Terminal output

```text id="vl9btx"
scons: Reading SConscript files ...
SCons 4.10.1 is using the following Python interpreter:
    /home/ischoegl/GitHub/cantera/.pixi/envs/default/bin/python (Python 3.14)

INFO: Building Cantera from git commit '4be08e69b'
INFO: Compiling on 'AMD Ryzen 9 9950X 16-Core Processor'
INFO: Configuration variables read from 'cantera.conf' and command line:

INFO: Adding conda include and library paths:
/home/ischoegl/GitHub/cantera/.pixi/envs/default

Checking for C++ header file cmath... yes
Checking whether __clang__ is declared... no

ERROR: Cantera requires a working implementation of 'std::isnan'.
If you have specified '-ffast-math' or equivalent as an optimization option,
either remove this option or add the '-fno-finite-math-only' option.

See 'config.log' for details.
```

[config.log](https://github.com/user-attachments/files/27805393/config.log)

Based on config.log, the issue is not actually about isnan. Instead, there are several linker errors trying to generate the executable for that test, of the form "undefined reference to `_dl_addr@GLIBC_PRIVATE'".

Besides whatever is causing this, our SCons logic could do a better job of distinguishing between compiler/linker errors at this point and a return value indicating problematic isnan behavior.

--- a/SConstruct
+++ b/SConstruct
@@ -313,11 +313,11 @@ config_options = [
     Option(
         "AR",
         "The archiver to use.",
-        "${AR}"),
+        os.environ.get("AR") or "${AR}"),
     Option(
         "CXX",
         "The C++ compiler to use.",
-        "${CXX}"),
+        os.environ.get("CXX") or "${CXX}"),
     Option(
         "cxx_flags",
         """Compiler flags passed to the C++ compiler only. Separate multiple
@@ -330,7 +330,7 @@ config_options = [
     Option(
         "CC",
         "The C compiler to use. This is only used to compile CVODE.",
-        "${CC}"),
+        os.environ.get("CC") or "${CC}"),
     Option(
         "cc_flags",
         """Compiler flags passed to both the C and C++ compilers, regardless of
@@ -397,10 +397,11 @@ config_options = [
         "default", ("y", "n", "default")),
     PathOption(
         "FORTRAN",
-        """The Fortran (90) compiler. If unspecified, the builder will look for a
-           compatible compiler (pgfortran, gfortran, ifort, ifx, g95) in the 'PATH'
-           environment variable. Used only for compiling the Fortran 90 interface.""",
-        "", PathVariable.PathAccept),
+        """The Fortran (90) compiler. If unspecified, the 'FC' environment variable is
+           used if set; otherwise the builder will look for a compatible compiler
+           (pgfortran, gfortran, ifort, ifx, g95) in the 'PATH' environment variable.
+           Used only for compiling the Fortran 90 interface.""",
+        os.environ.get("FC") or "", PathVariable.PathAccept),
     Option(
         "FORTRANFLAGS",
         "Compilation options for the Fortran (90) compiler.",

--- a/pixi.toml
+++ b/pixi.toml
@@ -31,8 +31,14 @@ pytest = ">=9.0.2,<10"
 "ruamel.yaml" = ">=0.19.1,<0.20"
 jinja2 = ">=3.1.6,<4"
 
+# Provide a self-consistent compiler toolchain matching the conda-forge-built
+# dependencies. The activation packages set CC/CXX/AR, which SConstruct honors. To build
+# with the system compiler instead, set `CXX = "g++"` and `CC = "gcc"` in cantera.conf.
 [feature.core.target.linux-64.dependencies]
-clangxx = ">=18.0"
+gxx_linux-64 = ">=13"
+
+[feature.core.target.linux-aarch64.dependencies]
+gxx_linux-aarch64 = ">=13"
 
 [feature.samples.dependencies]
 pandas = ">=3.0.0,<4"

